### PR TITLE
Report active experiments in error logs

### LINKED ISSUE #7484: Error Reporting should Record Experiments
Since we now allow various opt-ins, we should send the currently enabled experiments whenever an error happens.

/cc @cramforce, @dvoytenko

### PR DESCRIPTION
**Pull request by jridgewell**
Fixes #7484.

### CODE REVIEW

**Comment by dvoytenko** [comment]
/to @lannka given the related work recently.

File: src/error.js [hunk 1]
```diff
@@ -341,6 +341,9 @@ export function getErrorReportUrl(message, filename, line, col, error,
   }
   url += `&jse=${detectedJsEngine}`;
 
+  const experiments = experimentTogglesOrNull();
+  url += `&exps=${JSON.stringify(experiments)}`;
```
Comment (cramforce) [inline src/error.js]: This would need `encodeURIComponent`. Would a more compact custom encoding make sense?

**Comment by erwinmombay** [comment]
@ampprojectbot retry!

**Comment by erwinmombay** [comment]
@ampprojectbot retry!

File: src/error.js (hunk 1 above)
Comment (jridgewell) [inline src/error.js]: Done. I also moved away from JSON to a custom format.

File: src/experiments.js [hunk 2]
```diff
@@ -153,6 +127,32 @@ export function experimentToggles(win) {
   }
 
   Object.assign(toggles_, getExperimentTogglesFromCookie(win));
+
+  if (win.AMP_CONFIG
+      && Array.isArray(win.AMP_CONFIG['allow-url-opt-in'])
+      && win.AMP_CONFIG['allow-url-opt-in'].length > 0) {
+    const allowed = win.AMP_CONFIG['allow-url-opt-in'];
```
Comment (lannka) [inline src/experiments.js]: i don't see how this is related to the PR description..
is it a separate thing?

Comment (jridgewell) [inline src/experiments.js]: For me to test what's actually enabled on the page, I need a centralized location of the possible url enabled experiments. Imagine a bunch of `isExperimentOnAllowUrlOverride` calls in the source, how would we know?

Not to mention that if we could figure something out, we would need a dependency on `experiments.js` in the error reporter, which I'd rather avoid.

Comment (lannka) [inline src/experiments.js]: makes sense

**Pull request merged**